### ISSUE TITLE
Handle nginx error

### ISSUE THREAD

**Issue by ajayak** [nozzlegear/ShopifySharp#363]
Handle Shopify Error:

```
Unexpected character encountered while parsing value: <. Path '', line 0, position 0.: <html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.13.9</center>
</body>
```

**Comment by nozzlegear** [comment]
Good catch! I suppose the error parsing stuff should make sure the content type is JSON before trying to parse it.

**Comment by nozzlegear** [comment]
Fixed in 4.20.0, thanks for reporting this!
